freqtrade/persistence/models.py

```diff
@@ -590,9 +590,19 @@ def calc_close_trade_value(self, rate: Optional[float] = None,
         if rate is None and not self.close_rate:
             return 0.0
 
-        sell_trade = Decimal(self.amount) * Decimal(rate or self.close_rate)  # type: ignore
-        fees = sell_trade * Decimal(fee or self.fee_close)
-        return float(sell_trade - fees)
+        filled_exit_orders = self.select_filled_orders('sell')
+        filled_amount = 0
+        filled_exit_trade = 0
+        for order in filled_exit_orders:
+            filled_amount += order.filled
+            filled_exit_trade += (order.price * order.filled)
```

Comment:
Please use safe_filled / safe_price ... you can't 100% rely on filled being there.